### DISCUSSION TITLE
Unexpected token '<' and unityFramework is not defined

### DISCUSSION THREAD

**Discussion by haeunyy96** [jeffreylanters/react-unity-webgl#496] [Help Wanted!]
I am quite new to unity and I have been stuck here for a few days.

If someone could please help me that would be much appreciated... thank you!!!

![Screen Shot 2023-03-20 at 8 56 51 PM (2)](https://user-images.githubusercontent.com/107310198/226496179-3a30e989-4c41-4041-9f74-426d6c611648.png)

**Comment by UmerrAhsan** [comment]
Did you find the solution?

**Reply by haeunyy96** [reply]
I have not :(

**Reply by debbieyuen** [reply]
I had this problem with my student's projects. There were a couple of things that we had to fix on our end - though I'm unsure if these were the things that were causing the problem. 
1. Within Unity, make sure to hit "Disable" compression within the Player settings. The other modes apparently have been causing bugs. 
2. The builds must be clean builds. It did not work with a zipped build folder from itch.io and the files cannot be renamed. I recommend copying the entire build folder over including the index.html build file and the TemplateData folder. 
<img width="682" alt="Screenshot 2023-10-14 at 8 11 49 PM" src="https://github.com/jeffreylanters/react-unity-webgl/assets/31296177/54fddb2b-e517-4113-8ab5-fb2310155ba3">

3. For me, react-unity-webgl@8.x worked best. I needed to downgrade to 8.x

**Comment by Blondemonk** [comment]
`Unexpected token '<'` usually suggests that your server (or wherever you're loading your `build/*` files from) has returned a html response rather than the file you're after.

When you load the page in Chrome/Firefox, what does the Network tab show for each of the files you're loading? Do any of them load with the wrong mimetype or response?

In particular, in my UnityContext, my files don't have `.unityweb` at the end of them. If yours do, that's fine; however, if they don't then your server might not be able to find the right file. Even if your server is finding the files, it might not know what type they are (e.g. does it know that a `.js.unityweb` file should be served as `application/javascript`, and not something else).

**Reply by eastedgeZack** [reply]
![Screen Shot 2023-07-25 at 11 26 43 AM](https://github.com/jeffreylanters/react-unity-webgl/assets/98972054/d10cfc61-94a5-4ed1-b63c-549dfe7b1208)
![Screen Shot 2023-07-25 at 11 27 05 AM](https://github.com/jeffreylanters/react-unity-webgl/assets/98972054/c8019129-a49d-4657-99ba-3f4bd7ad0414)

Looks like it's trying to load the loader.js file as html but I'm not sure why

**Comment by eastedgeZack** [comment]
What unity version is this? I'm getting it on Unity 2021 and also 2022.3.5f1
![Screen Shot 2023-07-25 at 11 20 02 AM](https://github.com/jeffreylanters/react-unity-webgl/assets/98972054/ac6ae8be-a2ff-4a13-a433-da9ea59f9721)
![Screen Shot 2023-07-25 at 11 19 53 AM](https://github.com/jeffreylanters/react-unity-webgl/assets/98972054/85f4fe2e-6e3a-4a09-912f-20b38c42ce32)
![Screen Shot 2023-07-25 at 11 19 33 AM](https://github.com/jeffreylanters/react-unity-webgl/assets/98972054/22949e2b-0e82-42bc-ac44-7d8cf9ec9799)

**Comment by eastedgeZack** [comment]
Ah all I had to do was move the build folder to my public folder.
![Screen Shot 2023-07-25 at 12 07 08 PM](https://github.com/jeffreylanters/react-unity-webgl/assets/98972054/cbe23fca-c0a8-42e0-b3b7-1cd850fcc445)

**Reply by th-rutvik** [reply]
This works for me

**Reply by Igardemi** [reply]
It seems it was a permissions or cors problem, solved.
I believe that this problem is due to the cors permissions of the path, the resources are not found in those paths.
